main.go

```diff
@@ -29,19 +34,32 @@ func main() {
 	}
 	logrus.SetLevel(logLevel)
 
-	wg, err := newWGConfig(config.Interface, config.WireguardPort)
+	// Create the wireguard and cluster configuration
+	wgstate, err := wg.New(config.Interface, config.WireguardPort)
 	if err != nil {
 		logrus.WithError(err).Fatal("could not instantiate wireguard controller")
 	}
-
-	cluster, err := newCluster(config, wg)
+	cluster, err := cluster.New(config.Init, config.ClusterKey, config.BindAddr, config.ClusterPort, config.UseIPAsName)
 	if err != nil {
 		logrus.WithError(err).Fatal("could not create cluster")
 	}
 
-	nodec, errc := cluster.members() // avoid deadlocks by starting before join
+	// Prepare the /etc/hosts writer
+	hostsFile := &etchosts.EtcHosts{
+		Logger: logrus.StandardLogger(),
+	}
+
+	// Assign a local node address and propagate it to the cluster
+	wgstate.AssignOverlayAddr((*net.IPNet)(config.OverlayNet), cluster.LocalName)
+	localNode := common.Node{}
+	localNode.OverlayAddr = wgstate.OverlayAddr
+	localNode.PubKey = wgstate.PubKey.String()
+	cluster.Update(localNode)
```

Comment:
Something about this block feels off :thinking: 

How about this: unexport `AssignOverlayAddr()` and call it from `wg.New`, expanding it to also require the `cluster.LocalName` and `OverlayNet` (we can swap the instantiation order of `wgstate` and `cluster`, now that the latter doesn't require the former anymore).
Than we can add `wgstate.LocalNode()` that returns a `Node` with the appropriate info, and finally rename `cluster.Update` to something a bit more descriptive (maybe `Announce`? or `SetLocalNode`?)

Reply:
I would prefer a `wgstate.UpdateNode()` and creating the node in the main loop itself. It will allow for something else than the wireguard code to update the code, say the local routing table providing additional routes, as already asked in issue #28 .

Agreed otherwise, I will implement this right aways.